pytorch_lightning/callbacks/progress.py

```diff
@@ -20,18 +20,38 @@
 """
 import importlib
 import sys
+from numbers import Number
 
 # check if ipywidgets is installed before importing tqdm.auto
 # to ensure it won't fail and a progress bar is displayed
 from typing import Optional, Union
 
 if importlib.util.find_spec('ipywidgets') is not None:
-    from tqdm.auto import tqdm
+    from tqdm.auto import tqdm as _tqdm
 else:
-    from tqdm import tqdm
+    from tqdm import tqdm as _tqdm
 
 from pytorch_lightning.callbacks import Callback
 
+_PAD_SIZE = 5
+
+
+class tqdm(_tqdm):
+    """ Custom tqdm progressbar where we append 0 to floating points/strings to
+        prevent the progress bar from flickering
+    """
+    @staticmethod
+    def format_num(n: Number) -> str:
```

Comment:
I don't think Number is a good annotation, since this also could take strings and tensors...

Reply:
thanks, have removed it :]